### ISSUE TITLE
assume vertical soil C?

### ISSUE THREAD

**Issue by rgknox** [NGEET/fates#774]
With CTSM tag [ctsm5.1.dev054](https://github.com/ESCOMP/CTSM/releases/tag/ctsm5.1.dev054), the option to ignore layering of soil carbon was abandoned.   i.e. vertical stratification of soil bgc is enforced.

I'm wondering if there is any reason we would not want to make the same assumption.  This would allow us to assume that the relevant boundary conditions for C BGC, particularly heterotrophic respiration (in) and litter fluxes (out) would be aligned on the soil layering scheme. 

Note that when FATES is run, the namelist building scripts had been setting the use_vertsoilc flag active anyway.  Moreover, my sense is that FATES may already be making this assumption in various places, and I am very hesitant to think it would work correctly without layered soil carbon/bgc boundary conditions.

TLDR: It may be a good idea to formally remove the input flag specifying the vertical soil carbon scheme and just assume layered:

https://github.com/ESCOMP/CTSM/pull/1182/files/74efe8a8822fce67f4171e51e0251b3f5f4d5fe5..c642c2c828e3a39f4c0fb446f6db8af3da76c04c#diff-a9a4094690743c218e0d8579a71f890acc0a50793c38f0f4a9981fc2914935b6R338

**Comment by ckoven** [comment]
Is E3SM now also always running with vertical soil carbon?

**Comment by rgknox** [comment]
I don't think so, but we could impose that it does

**Comment by glemieux** [comment]
Per discussion with @ckoven it would be better to remove this line here:
https://github.com/NGEET/fates/blob/aa587ccf3a94abbd68dbf3dc7646fa0a61e411ba/main/FatesInterfaceMod.F90#L461

To force an `endrun` when `use_vertsoil` is false.
